docker-images/operator/scripts/launch_java.sh

```diff
@@ -24,8 +90,13 @@ JAVA_OPTS="${JAVA_OPTS} $(get_gc_opts)"
 JAVA_OPTS="${JAVA_OPTS} --illegal-access=deny"
 
 # Default memory options used when the user didn't configured any of these options, we set the defaults
-if [[ "$JAVA_OPTS" != *"MinRAMPercentage"* && "$JAVA_OPTS" != *"MaxRAMPercentage"* && "$JAVA_OPTS" != *"InitialRAMPercentage"* ]]; then
+if [[ ! -r "${mem_file_cgroups_v2}" && "$JAVA_OPTS" != *"MinRAMPercentage"* && "$JAVA_OPTS" != *"MaxRAMPercentage"* && "$JAVA_OPTS" != *"InitialRAMPercentage"* ]]; then
   JAVA_OPTS="${JAVA_OPTS} -XX:MinRAMPercentage=10 -XX:MaxRAMPercentage=20 -XX:InitialRAMPercentage=10"
+elif [[ -r "${mem_file_cgroups_v2}" && "$JAVA_OPTS" != *"Xmx"* ]]; then
+  # Set env vars reflecting memory limits
+  init_memory_limit_env_vars
+  # Calculate -Xmx java option
+  JAVA_OPTS="${JAVA_OPTS} $(calc_max_memory)"
 fi
```

Comment:
Perhaps I'm missing something, but the else code is only executed if the pod is deployed in a cluster using cgroups v2 (XX:MinRAMPercentage , XX:MinRAMPercentage, don't work in that case) and the user does not specify -Xmx.

Since we're going to simplify the process of getting that value, calc_max_memory should always return a value specified in /sys/fs/cgroup/memory.max

What do you think?